src/main/webapp/query-control.js

```diff
@@ -290,6 +290,56 @@ function replaceValueIfEmpty(dataListId) {
   }
 }
 
+// When the user picks certain dropdown values,
+// exclude the combinations that don't have data available
+const dropdownsToExclude = new Map();
+dropdownsToExclude.set(
```

Comment:
the same values have to be listed twice once for persontype to action and once again for action to persontype. You could instead make this just a set of personType+action. Then when checking you compute personType+action and check if it exists in the set.

Reply:
Ah, good idea. I've done this, hopefully in a way that's not too clunky.